tools/ci_build/github/azure-pipelines/templates/mac-build-step-with-cache.yml

```diff
@@ -0,0 +1,41 @@
+# It's used for compilation with cache in Mac Host.
+
+parameters:
+- name: TODAY
+  type: string
+
+- name: BuildStep
+  type: stepList
+
+- name: AdditionalKey
+  type: string
+
+- name: CACHE_DIR
+  type: string
+
+steps:
+  - script: |
+      brew install ccache
+      echo "##vso[task.prependpath]/usr/local/opt/ccache/libexec"
+      mkdir -p "${{ parameters.CACHE_DIR }}"
+    displayName: Install ccache and update PATH to use linked versions of gcc, cc, etc
+
+  - task: Cache@2
+    inputs:
+      key:  ' "${{parameters.TODAY}}" | ${{parameters.AdditionalKey}} | "$(Build.SourceVersion)" '
+      path: ${{ parameters.CACHE_DIR }}
+      restoreKeys: |
+        "${{parameters.TODAY}}" | ${{parameters.AdditionalKey}}
+    displayName: ccache task
+
+  - ${{ parameters.BuildStep }}
+
+  - script: |
+      set -ex
+      ccache -s
+      ccache -z
+      ls -l $(Pipeline.Workspace)/ccache
+      du -sh $(Pipeline.Workspace)/ccache
+    displayName: Show Cache stats and Clear stats.
```

Comment:
do we also want to run this step if the build steps are not successful?

Reply:
If the build steps are no successful, the workflow will be broken, and it is impossible to run the step.
This step is to print some logs to investigate cache issue.

Reply:
I think the question is whether this step would provide valuable info and/or be useful in the case of a failure, and as such you should set a condition to run always.

e.g. could a build failure leave the cache in an invalid state so we always want to clear it.

Reply:
No, it doesn't provide useful info if there's a failure.
The most import info is the cache hit rate.
On CI machine, since the directory of workspace would be clear after one job, it's not necessary to run` ccache -z` at the end.
In #15267, I wrote a template function for display info. 
The ` ccache -z` is useful for next cache task in the same pipeline.

Reply:
It looks it's no harm to run it always. I added the condition.